system-int-test/test/cmr/system_int_test/ingest/collection_misc_validation_test.clj

```diff
@@ -77,6 +78,43 @@
       (is (some? (re-find #"/Purpose string.*is too long \(length: 12000, maximum allowed: 10000\)" (:warnings ingest-response))))
       (is (some? (re-find #"/Purpose string.*is too long \(length: 12000, maximum allowed: 10000\)" (:warnings validation-response)))))))
 
+(deftest error-messages-are-friendly
+  (testing "Error messages don't contain regexes"
+    ;; Include fields that will fail the regex pattern and ensure that the regexes
+    ;; don't show up in the returned errors
+    (let [platform-collection (data-umm-c/collection
+                               {:Platforms [(data-umm-cmn/platform {:ShortName ""
+                                                                    :LongName ""})]})
+          data-center-collection (data-umm-c/collection
+                                  {:DataCenters [(data-umm-cmn/data-center {:ShortName "asdf"
+                                                                            :Roles ["ARCHIVER"]})]})
+          science-keyword-collection (data-umm-c/collection
+                                      {:ScienceKeywords
+                                       [(data-umm-cmn/science-keyword
+                                         {:VariableLevel1 ""
+                                          :VariableLevel2 ""
+                                          :VariableLevel3 ""
+                                          :Category ""
+                                          :Term ""
+                                          :DetailedVariable ""
+                                          :Topic ""})]})]
+
+      (are3 [collection]
+       (let [ingest-response (d/ingest-umm-spec-collection "PROV1" collection {:format :umm-json :allow-failure? true})
+             validation-response (ingest/validate-concept (data-umm-c/collection-concept collection :umm-json))]
+         (is (nil? (first (map #(re-find #"(ECMA|regex)" %) (:errors ingest-response)))))
+         (is (nil? (first (map #(re-find #"(ECMA|regex)" %) (:errors validation-response))))))
+       "Invalid platforms"
+       platform-collection
+
+       "Invalid Data Centers"
+       data-center-collection
+
+       "Invalid Science Keywords"
+       science-keyword-collection))))
+
+
```

Comment:
I want an explicit test added that verifies the error message returned is:

`/Platforms/1/LongName string "" is too short (length: 0, required minimum: 1)`

rather than:
`/Platforms/1/LongName string "" is too short (length: 0, required minimum: 1); /Platforms/1/LongName ECMA 262 regex "[\w\-&'()\[\]/."#$%\^@!*+=,][\w-&'”)`

Reply:
So now I'm saying instead:
```
/Platforms/1/LongName string "" is too short (length: 0, required minimum: 1);
/Platforms/1/LongName string is in an invalid format.
```